extensions/ql-vscode/src/variant-analysis/view-autofixes.ts

```diff
@@ -0,0 +1,801 @@
+import type { RepositoriesFilterSortStateWithIds } from "./shared/variant-analysis-filter-sort";
+import {
+  defaultFilterSortState,
+  filterAndSortRepositoriesWithResults,
+} from "./shared/variant-analysis-filter-sort";
+import type {
+  VariantAnalysis,
+  VariantAnalysisRepositoryTask,
+} from "./shared/variant-analysis";
+import type { Credentials } from "../common/authentication";
+import type { NotificationLogger } from "../common/logging";
+import type { App } from "../common/app";
+import type { CodeQLCliServer } from "../codeql-cli/cli";
+import { pathExists, ensureDir, readdir, move, remove } from "fs-extra";
+import { withProgress, progressUpdate } from "../common/vscode/progress";
+import type { ProgressCallback } from "../common/vscode/progress";
+import { join, dirname, parse } from "path";
+import { tryGetQueryMetadata } from "../codeql-cli/query-metadata";
+import { window as Window } from "vscode";
+import { pluralize } from "../common/word";
+import { glob } from "glob";
+import { readRepoTask } from "./repo-tasks-store";
+import { unlink, mkdtemp, readFile, writeFile } from "fs/promises";
+import { tmpdir } from "os";
+import { spawn } from "child_process";
+import type { execFileSync } from "child_process";
+import { tryOpenExternalFile } from "../common/vscode/external-files";
+
+// Limit to three repos when generating autofixes so not sending
+// too many requests to autofix. Since we only need to validate
+// a handle of autofixes for each query, this should be sufficient.
+// Consider increasing this in the future if needed.
+const MAX_NUM_REPOS: number = 3;
+// Similarly, limit to three fixes per repo.
+const MAX_NUM_FIXES: number = 3;
+
+/**
+ * Generates autofixes for the results of a variant analysis.
+ */
+export async function viewAutofixesForVariantAnalysisResults(
+  variantAnalysisId: number,
+  filterSort: RepositoriesFilterSortStateWithIds = defaultFilterSortState,
+  variantAnalyses: Map<number, VariantAnalysis>,
+  credentials: Credentials,
+  logger: NotificationLogger,
+  storagePath: string,
+  app: App,
+  cliServer: CodeQLCliServer,
+): Promise<void> {
+  await withProgress(
+    async (progress: ProgressCallback) => {
+      // Get the path to the local autofix installation.
+      progress(progressUpdate(1, 4, `Checking for local autofix installation`));
```

Comment:
Since the `findLocalAutofix` function should be very fast, does this actually show up in the progress window?

Reply:
Good point. I don't think I've seen it show up. Same for the `overrideQueryHelp` progress update. I've removed them in https://github.com/github/vscode-codeql/pull/4065/commits/7d3e403e0b2b158fa4e8732123b58600b360f295.